Analysis/interface/GenLepton.h

```diff
@@ -243,8 +243,15 @@ class GenLepton {
             size_t last_mother_index = NoneIndex;
 
             if(!particle->motherRefVector().empty()) {
-                for(const auto& mother : particle->motherRefVector())
+		int Nmother = 0;
+                for(const auto& mother : particle->motherRefVector()) {
                     FillDaughters(mother.get(), NoneIndex, false);
+		    ++Nmother;
+		    if(Nmother>6) {
+			    std::cout<<"Warning : Gen particle with > 6 mothers, storing only 6 firsts"<<std::endl;
+			    break;
+		    }
```

Comment:
May I ask: in which MC it is observed? Is it for pp interaction level summing up over all the partons? I am a bit concerned why do we want to save it

Reply:
Having more than 2 mothers occur regularly in all our MC in tau decays from FSR. The total number of mothers in fragmentation does not have an upper limit (from physics perspective). For technical reasons, we can save only up to 6 mothers. But we don't need to have the precise decay graph for FSR products, so I think it is ok to store only first 6 mothers.